### PR TITLE
Add observedGeneration field to CR status

### PR DESCRIPTION
**Pull request by kabicin**
https://github.com/OpenLiberty/open-liberty-operator/issues/535 
- Updates `.status.observedGeneration` at the end of the reconcile loop by copying `.metadata.generation`

### CODE REVIEW

File: api/v1/openlibertyapplication_types.go [hunk 1]
```diff
@@ -147,6 +147,9 @@ type OpenLibertyApplicationSpec struct {
 
 	// +operator-sdk:csv:customresourcedefinitions:order=32,type=spec,displayName="Topology Spread Constraints"
 	TopologySpreadConstraints *OpenLibertyApplicationTopologySpreadConstraints `json:"topologySpreadConstraints,omitempty"`
+
+	// +operator-sdk:csv:customresourcedefinitions:order=33,type=spec,displayName="Generation"
+	Generation *int32 `json:"generation,omitempty"`
```
Comment (leochr) [inline api/v1/openlibertyapplication_types.go:152]: Let's use the generation field under metadata of the CR, which is automatically incremented when the CR's spec changes. That'll avoid user having to change the field manually.

File: api/v1/openlibertyapplication_types.go [hunk 2]
```diff
@@ -432,6 +435,8 @@ type OpenLibertyApplicationStatus struct {
 	References common.StatusReferences `json:"references,omitempty"`
 
 	SemeruCompiler *SemeruCompilerStatus `json:"semeruCompiler,omitempty"`
+
+	ObservedGeneration *int32 `json:"observedGeneration,omitempty"`
```
Comment (leochr) [inline api/v1/openlibertyapplication_types.go:439]: Please include a description for the field

**Review by leochr — COMMENTED** [review]
@kabicin Thanks for the PR. Added some comments

File: api/v1/openlibertyapplication_types.go (hunk 1 above)
Comment (kabicin) [reply api/v1/openlibertyapplication_types.go]: Removed this extra field

File: api/v1/openlibertyapplication_types.go (hunk 2 above)
Comment (kabicin) [reply api/v1/openlibertyapplication_types.go]: I added a description and changed the variable type to int64 to match .metadata.generation.

**Review by leochr — COMMENTED** [review]
@kabicin Thanks for the PR. Looks good. Just one comment:
We must have missed adding Reconciled version after [this line](https://github.com/OpenLiberty/open-liberty-operator/pull/554/files#diff-a00ebb108908f3e8e9540b617814aa8f44208ec2d0aa63fd48ac65e3b403a185R123) previously. Please add the Reconciled version and observedGeneration to keep in sync with WLO. Thanks.

**Review by leochr — APPROVED** [review]

**Pull request merged**